### PR TITLE
Add tslint, a Typescript Linter.

### PR DESCRIPTION
**Pull request by BramGruneir**
This adds a linter for Typescript ... and fixed all the lints.

To add it locally, just run 
npm install -g tslint

### CODE REVIEW

File: resource/Makefile [hunk 1]
```diff
@@ -43,6 +43,7 @@ debug: $(GOBINDATA_DEBUG_TARGET)
 
 $(TYPESCRIPT_TARGET): $(shell find ui/ts)
 	tsc -p ui/ts
+	tslint -c ui/ts/tslint.json $(addprefix -f ,$(shell find ui/ts -name *.ts -not -path *typings*))
```
Comment (tamird) [inline resource/Makefile]: this comma has a space on the wrong side

**Comment by tamird** [comment]
LGTM but i didn't review the TS

File: resource/Makefile (hunk 1 above)
Comment (BramGruneir) [inline resource/Makefile]: fixed

**Comment by mrtracy** [comment]
Most of the changes are very agreeable, the only one I feel strongly about is the type requirement on variables where the type can be clearly inferred from the assignment.  This ability has actually been *added* to many languages such as C++ through the auto keyword; our backend language also has this ability.  It seems overly strict to disallow it here.

**Comment by BramGruneir** [comment]
It's funny, the linter I used to use for C# actually enforced using the `var` keyword, which was added to clean up the code.  I'll disable that rule, but I'm not going to go back through all the code to fix the ones that I've changed.

**Comment by BramGruneir** [comment]
@mrtracy, can you take a final look?  I'd like to merge this.

**Comment by mrtracy** [comment]
LGTM

**Pull request merged**